metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/WorkflowExecutionDao.java

```diff
@@ -284,7 +284,7 @@ public String existsAndNotCompleted(String datasetId) {
    *                    values.
    * @return the first plugin found
    */
-  public PluginWithExecutionId<MetisPlugin> getFirstSuccessfulPlugin(String datasetId,
+  public PluginWithExecutionId<MetisPlugin<?>> getFirstSuccessfulPlugin(String datasetId,
```

Comment:
That's an interesting point, as I did some of this to get rid of some warnings showing up in my IDE. But some others have now appeared, so it didn't quite work:)
There was another reason though, I got an untyped cast exception somewhere, which set me on the path to work on this in the first place.
I decided to keep it this way because it forces us to think more about the typing in our code, but I didn't expect that it would affect our sonarQube. What do you think?

Reply:
Actually, perhaps I can do some cleverness with untyped interfaces to solve this issue.

Reply:
I think that the current sonarqube issue is "worse" than the previous issue.. What I mean is, I think before it was more an IDE issue while now it has become a "Major code smell".
I'm okay with both tbh but we should take action for at least the latter if we leave as is.
For the scope if this ticket I guess we can move forward, but we should try to take action with the idea you had as a follow up, and/or somehow mark those cases in Sonar.
My first intuition is to do an immediate follow up with your idea, what do you think?

Reply:
Sure, sounds good. You're right, I don't think we should keep it this way, and I'd like to find the nicest way to solve this. If I have time this sprint I will do a tech debt ticket on this. But given the time constraints, it is not currently part of this ticket. Does that sound acceptable?

Reply:
Yeah sounds good!